pkg/platform/kube/monitoring/function.go

```diff
@@ -164,29 +164,42 @@ func (fm *FunctionMonitor) updateFunctionStatus(function *nuclioio.NuclioFunctio
 	return nil
 }
 
-// consider function deployment available if it has at least one available pod
+// consider function deployment available when all pods are available
 func (fm *FunctionMonitor) isAvailable(deployment *appsv1.Deployment) bool {
-	return !(deployment.Spec.Replicas != nil &&
+	atLeastOneRunningPod := !(deployment.Spec.Replicas != nil &&
 		*deployment.Spec.Replicas > 0 &&
 		deployment.Status.AvailableReplicas <= 0)
-
+	allReplicasAvailable := deployment.Status.UnavailableReplicas <= 0
+	return atLeastOneRunningPod && allReplicasAvailable
 }
 
+// We monitor functions that meet the following conditions:
+// 1. not in provisioning state
+// 2. not recently deployed
+// 3. not in transitional states
 func (fm *FunctionMonitor) shouldSkipFunctionMonitoring(function *nuclioio.NuclioFunction) bool {
 
-	// ignore provisioning states
-	// ignore recently deployed function
+	// (1) ignore provisioning states
+	// (2) ignore recently deployed function
 	if fm.resolveFunctionProvisionedOrRecentlyDeployed(function) {
 		fm.logger.DebugWith("Skipping check for function status change",
 			"functionName", function.Name)
 		return true
 	}
 
-	// ignore transitional states other than ready / error
-	return !functionconfig.FunctionStateInSlice(function.Status.State, []functionconfig.FunctionState{
+	// (3) ignore transitional states other than ready / error
+	if !functionconfig.FunctionStateInSlice(function.Status.State, []functionconfig.FunctionState{
 		functionconfig.FunctionStateReady,
-		functionconfig.FunctionStateError,
-	})
+		functionconfig.FunctionStateUnhealthyError,
+	}) {
```

Comment:
I think it would be better to inclusively state the transitional states here. More stable

Reply:
I disagree in this case. adding a state would without taking care of it here would lead to function monitoring handle functions in such state.

moreover, it is much easier to read it that way, e.g.: "non-transitional state is everything but X Y Z states" as oppose to "non-transitional is not one of A B C states" makes it much easier to know that non-transitional could be one of X Y Z states.

Reply:
What are transitional states here? All the provisioning states are handled. 
You wrote "(3) ignore transitional states other than ready / error":
- error -> unhealthy
- ready and unhealthy states are not transitional,or are they?
What are the transitional states that you are skipping here? Specifically?

Reply:
comment was bad - it should have been
`ignore provisioned states other than ready / unhealthy`

so that means, if your function is provisioned and not ready/unhealthy - do not monitor.